src/servant/tt_001/F84.c

```diff
@@ -57,7 +77,254 @@ Entity* func_us_80171568(Entity* self, s32 entityId) {
+                self->hitboxWidth = 8;
+                self->hitboxHeight = 8;
+            }
+        } else {
+            self->ext.ghost.unk86 = 0;
+            if (self->ext.ghost.unk92->entityId == 0xDA) {
+                self->ext.ghost.unk92->params = 1;
+                D_us_80170028(self->ext.ghost.unk92);
+            }
+        }
+        self->posX.val += self->velocityX;
+        self->posY.val += self->velocityY;
+        break;
+    case 4:
+        if (!(g_Player.unk0C & 0x01000001)) {
+            if (self->ext.ghost.unk96->entityId == 0xDB) {
+                self->ext.ghost.unk96->params = 1;
+                D_us_8017002C(self->ext.ghost.unk96);
+            }
+            self->step = 1;
+            break;
+        }
+        func_us_80170F84(self, D_us_801737C4, D_us_801737C8);
+        self->posY.val += self->velocityY;
+        switch (self->ext.ghost.unk8C) {
+        case 0:
+            self->ext.ghost.unk86 = 0;
+            self->ext.ghost.unk8C++;
+            break;
+        case 1:
+            if (++self->ext.ghost.unk86 >= 0x3D) {
+                self->ext.ghost.unk86 = 0;
+                self->ext.ghost.unk8C++;
+            }
+            break;
+        case 2:
+        case 3:
+        case 4:
+        case 5:
+            self->ext.ghost.unk86++;
+            if (self->ext.ghost.unk86 == 1) {
+                self->facingLeft = self->facingLeft == 0;
+            } else if (self->ext.ghost.unk86 >= 0x10) {
+                self->ext.ghost.unk86 = 0;
+                self->ext.ghost.unk8C++;
+            }
+            break;
+        case 6:
+            if (++self->ext.ghost.unk86 >= 0x3D) {
+                self->ext.ghost.unk86 = 0;
+                self->ext.ghost.unk8C++;
+            }
+            break;
+        case 7:
+            self->ext.ghost.unk86++;
+            if (self->ext.ghost.unk86 == 1) {
+                self->ext.ghost.unk96 = func_us_80171568(self, 1);
+            } else if (self->ext.ghost.unk86 >= 0x3D) {
+                self->ext.ghost.unk86 = 0;
+                self->step++;
+            }
+            break;
+        }
+        break;
+    case 5:
+        if (g_Player.unk0C & 0x01000001) {
+            if (self->velocityY > (s32)0xFFFE8000) {
+                self->velocityY -= FIX(0.25);
+            }
+            self->posY.val += self->velocityY;
+            if (self->posY.i.hi < -0x10) {
+                self->velocityY = 0;
+                self->posX.val = FIX(128.0);
+                self->posY.val = FIX(-16.0);
+                self->step++;
+            }
+        } else {
+            self->step = 1;
+        }
+        break;
+    case 6:
+        if (!(g_Player.unk0C & 0x01000001)) {
+            self->step = 1;
+        } else {
+            self->posX.val = FIX(128.0);
+            self->posY.val = FIX(-16.0);
+        }
+        break;
+    }
+    self->ext.ghost.unk88 += self->ext.ghost.unk8A;
+    if ((s16)(self->ext.ghost.unk88 - 0x21) >= 0x5Fu) {
```

Comment:
Something like
```
    if(!(arg0->ext.ghost.unk88 >= 0x21 && arg0->ext.ghost.unk88 < 0x5F + 0x21))
```
haven't verified on psp

Reply:
Yeah, that change cleans up a lot on the PSP version as well.